### PR TITLE
Adding fearless FFI article

### PR DESCRIPTION
**Pull request by Verdagon**

### CODE REVIEW

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+This is part of Vale's goal to be the *safest native language* [# By native, we mean not running in a VM.] Most languages compromise memory safety in some way, which can lead to difficult bugs, security vulnerabilities, and things like supply chain attacks.
+
+
+However, Vale takes a big step forward here by isolating unsafe and untrusted code, and keeping it from undermining the safe code around it.
+
+
+This page describes the proof-of-concept we have so far, plus the next steps. It involves some borderline-insane acrobatics with *inline assembly*, *bitwise xor and rotate*, and *two simultaneous stacks*. Buckle up!
+
+
+If you're impressed with our [track record so far](https://vale.dev/roadmap) and believe in the direction we're heading, please consider [sponsoring us on github](https://github.com/sponsors/ValeLang)! We can't do this without you, and we appreciate all the support you've shown.
```
Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: GitHub

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+>>>>
+
+
+
+## References Between C and Vale
+
+Message passing is great for most use cases, but we also want to be able to hand C some pointers to our Vale data. That way, C code can call methods on our Vale objects.
+
+
+For example, we might want to wrap a Rust HTTP server and have it call into Vale to handle requests. [# I did this recently in [RocketVale](https://github.com/verdagon/rocketvale) to implement a game server for a [roguelike game](https://verdagon.itch.io/vale-prototype).]
```
Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: Inconsistent to mention Rust here, why not stick to C?

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+As we implement the sandboxing portion, we'll be posting follow-up articles about the details and tradeoffs involved here. Stay tuned!
+
+
+# Putting it all together
+
+We've described four mechanisms to help protect our Vale data from problems in our C code:
+
+ * Message passing. (Done!)
+ * Scrambled references. (Done!)
+ * Separate stacks. (Prototyped in Mac OS!)
```
Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: macOS

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+So far, these mechanisms protect us from *accidental* corruption. With the system above, it's pretty much impossible to accidentally corrupt Vale objects. With this, projects or teams can call into their own C code, and have confidence that they aren't causing memory unsafety in their Vale objects.
+
+
+This system also protects us from accidental memory safety in our dependencies. Dependencies' unsafety is a big problem in projects nowadays.
+
+
+With these mechanisms, we can be confident that any problems in our dependencies won't cause memory unsafety in our code.
+
+
+*We implemented a proof-of-concept of this!* The proof of concept works for Mac OS; simply supply `--enable_side_calling true` to the `valec` invocation. The other measures (message passing, references, scrambling) already work on all platforms and are enabled by default.
```
Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: Mac OS is stylized by apple as macOS

Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: `;` seems pointless, "The proof of concept works for Mac OS by supplying `--enable_side_calling true` to the `valec` invocation"

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+realm: blog
+path: blog/fearless-ffi
+layout: annotated
+namespace: c-blog m-annotated
+---
+
+[Vale 0.2](/blog/version-0.2-released) is out, and it includes the beginnings of a feature we like to call *Fearless FFI*.
+
+
+This is part of Vale's goal to be the *safest native language* [# By native, we mean not running in a VM.] Most languages compromise memory safety in some way, which can lead to difficult bugs, security vulnerabilities, and things like supply chain attacks.
```
Comment (jhburns) [inline src/blog/0.2/next-fearless-ffi.vmd]: Missing `.`

File: src/blog/0.2/next-fearless-ffi.vmd [hunk 6]
```diff
@@ -0,0 +1,428 @@
+---
```
Comment (Verdagon) [inline src/blog/0.2/next-fearless-ffi.vmd]: Add example of a supply chain attack

Comment (Verdagon) [inline src/blog/0.2/next-fearless-ffi.vmd]: "Copying" or "Deep copying" instead of "message passing"

**Pull request closed without merging**